diff --git a/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java b/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
index 4d2e1a4e9a..78b523e350 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
@@ -2274,6 +2274,9 @@ public Void call() throws Exception {
                   + " isAcid=" + isAcid + ", "
                   + " hasFollowingStatsTask=" + hasFollowingStatsTask, t);
               throw t;
+            } finally {
+              // get(conf).getMSC can be called in this task, Close the HMS connection right after use, do not wait for finalizer to close it.
+              closeCurrent();
             }
           }
         }));
