diff --git a/ambari-web/app/controllers/main/admin/highAvailability/nameNode/step7_controller.js b/ambari-web/app/controllers/main/admin/highAvailability/nameNode/step7_controller.js
index 722668cf05..27ea705e48 100644
--- a/ambari-web/app/controllers/main/admin/highAvailability/nameNode/step7_controller.js
+++ b/ambari-web/app/controllers/main/admin/highAvailability/nameNode/step7_controller.js
@@ -32,11 +32,11 @@ App.HighAvailabilityWizardStep7Controller = App.HighAvailabilityProgressPageCont
       tasksToRemove.push('startAmbariInfra');
     }
 
-    if (App.ClientComponent.getModelByComponentName('RANGER_ADMIN').get('installedCount') === 0) {
+    if (App.ClientComponent.getModelByComponentName('RANGER_ADMIN') == null || App.ClientComponent.getModelByComponentName('RANGER_ADMIN').get('installedCount') === 0) {
       tasksToRemove.push('startRanger');
     }
 
-    if (App.ClientComponent.getModelByComponentName('MYSQL_SERVER').get('installedCount') === 0) {
+    if (App.ClientComponent.getModelByComponentName('MYSQL_SERVER') == null || App.ClientComponent.getModelByComponentName('MYSQL_SERVER').get('installedCount') === 0) {
       tasksToRemove.push('startMysqlServer');
     }
 
