diff --git a/ambari-metrics/ambari-metrics-grafana/conf/unix/ambari-metrics-grafana b/ambari-metrics/ambari-metrics-grafana/conf/unix/ambari-metrics-grafana
index dba2f5ff55..58c2598208 100644
--- a/ambari-metrics/ambari-metrics-grafana/conf/unix/ambari-metrics-grafana
+++ b/ambari-metrics/ambari-metrics-grafana/conf/unix/ambari-metrics-grafana
@@ -118,7 +118,7 @@ case "$1" in
     then
       for i in {1..10}
       do
-        sleep 2
+        sleep 12
         # check if pid file has been written to
         if [ -s $PID_FILE ]; then
           echo " $(date) pid_file has been written to" >> $LOG_FILE
@@ -132,7 +132,7 @@ case "$1" in
         exit 1
       fi
       i=0
-      timeout=60
+      timeout=120
       # Wait for the process to be properly started before exiting
       until { cat "$PID_FILE" | xargs kill -0; } >/dev/null 2>&1
       do
