diff --git a/ambari-agent/src/main/package/dependencies.properties b/ambari-agent/src/main/package/dependencies.properties
index 07b0b68319..9cfc2daeb5 100644
--- a/ambari-agent/src/main/package/dependencies.properties
+++ b/ambari-agent/src/main/package/dependencies.properties
@@ -28,5 +28,5 @@
 # Such a format is respected by install_ambari_tarball.py by default, 
 # however should be encouraged manually in pom.xml.
 
-rpm.dependency.list=openssl,\nRequires: rpm-python,\nRequires: zlib,\nRequires: python >= 2.6
-deb.dependency.list=openssl, zlibc, python (>= 2.6)
\ No newline at end of file
+rpm.dependency.list=openssl,\nRequires: zlib,\nRequires: python2 >= 2.6
+deb.dependency.list=openssl, zlibc, python (>= 2.6)
diff --git a/ambari-common/src/main/python/resource_management/core/providers/package/__init__.py b/ambari-common/src/main/python/resource_management/core/providers/package/__init__.py
index 799cf3eed0..ade58292cb 100644
--- a/ambari-common/src/main/python/resource_management/core/providers/package/__init__.py
+++ b/ambari-common/src/main/python/resource_management/core/providers/package/__init__.py
@@ -359,9 +359,8 @@ class RPMBasedPackageProvider(PackageProvider):
   INSTALLED_PACKAGE_VERSION_COMMAND = "rpm -q --queryformat '%{{version}}-%{{release}}' \"{0}\""
 
   def rpm_check_package_available(self, name):
-    import rpm # this is faster then calling 'rpm'-binary externally.
-    ts = rpm.TransactionSet()
-    packages = ts.dbMatch()
+    import os
+    packages = os.popen("rpm -qa --queryformat '%{name} '").read().split()
 
     name_regex = re.escape(name).replace("\\?", ".").replace("\\*", ".*") + '$'
     regex = re.compile(name_regex)
diff --git a/ambari-common/src/main/python/resource_management/core/providers/package/yumrpm.py b/ambari-common/src/main/python/resource_management/core/providers/package/yumrpm.py
index da2a3b1de1..503644137c 100644
--- a/ambari-common/src/main/python/resource_management/core/providers/package/yumrpm.py
+++ b/ambari-common/src/main/python/resource_management/core/providers/package/yumrpm.py
@@ -305,10 +305,7 @@ class YumProvider(RPMBasedPackageProvider):
     yum in inconsistant state (locked, used, having invalid repo). Once packages are installed
     we should not rely on that.
     """
-    if os.geteuid() == 0: 
-      return self.yum_check_package_available(name)
-    else:
-      return self.rpm_check_package_available(name)
+    return self.rpm_check_package_available(name)
 
   def yum_check_package_available(self, name):
     """
diff --git a/ambari-server/src/main/package/dependencies.properties b/ambari-server/src/main/package/dependencies.properties
index 01244a8375..70205d5dc6 100644
--- a/ambari-server/src/main/package/dependencies.properties
+++ b/ambari-server/src/main/package/dependencies.properties
@@ -28,6 +28,6 @@
 # Such a format is respected by install_ambari_tarball.py by default, 
 # however should be encouraged manually in pom.xml.
 
-rpm.dependency.list=postgresql-server >= 8.1,\nRequires: openssl,\nRequires: python >= 2.6
+rpm.dependency.list=postgresql-server >= 8.1,\nRequires: openssl,\nRequires: python2 >= 2.6
 rpm.dependency.list.suse=postgresql-server >= 8.1,\nRequires: openssl,\nRequires: python-xml,\nRequires: python >= 2.6
-deb.dependency.list=openssl, postgresql (>= 8.1), python (>= 2.6), curl
\ No newline at end of file
+deb.dependency.list=openssl, postgresql (>= 8.1), python (>= 2.6), curl
