diff --git a/ambari-common/src/main/python/resource_management/libraries/providers/hdfs_resource.py b/ambari-common/src/main/python/resource_management/libraries/providers/hdfs_resource.py
index 301f2b8b63..d6bf6e6c36 100644
--- a/ambari-common/src/main/python/resource_management/libraries/providers/hdfs_resource.py
+++ b/ambari-common/src/main/python/resource_management/libraries/providers/hdfs_resource.py
@@ -643,7 +643,7 @@ class HdfsResourceProvider(Provider):
     else:
       path = path
       
-    return re.sub("[/]+", "/", path)
+    return re.sub("[/]+", "/", path).replace(" ", "%20")
 
   @staticmethod
   def get_ignored_resources_list(hdfs_resource_ignore_file):
