--- oozie-4.2.0/core/src/main/java/org/apache/oozie/dependency/HCatURIHandler.java
+++ oozie-4.2.0/core/src/main/java/org/apache/oozie/dependency/HCatURIHandler.java
@@ -48,6 +48,9 @@ import org.apache.oozie.service.Services;
 import org.apache.oozie.service.URIHandlerService;
 import org.apache.oozie.util.HCatURI;
 import org.apache.oozie.util.XLog;
+import org.apache.hadoop.hive.thrift.DelegationTokenIdentifier;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.security.token.Token;

 public class HCatURIHandler implements URIHandler {

@@ -270,8 +273,10 @@ public class HCatURIHandler implements URIHandler {
                     delegationToken = tokenClient.getDelegationToken(user, UserGroupInformation.getLoginUser()
                             .getUserName());
                     // Store Delegation token in the UGI
-                    ShimLoader.getHadoopShims().setTokenStr(ugi, delegationToken,
-                            hiveConf.get("hive.metastore.token.signature"));
+                    Token<DelegationTokenIdentifier> token = new Token<DelegationTokenIdentifier>();
+                    token.decodeFromUrlString(delegationToken);
+                    token.setService(new Text(hiveConf.get("hive.metastore.token.signature")));
+                    ugi.addToken(token);
                 }
                 finally {
                     if (tokenClient != null)
--- oozie-4.2.0/core/src/test/java/org/apache/oozie/test/hive/MiniHS2.java
+++ oozie-4.2.0/core/src/test/java/org/apache/oozie/test/hive/MiniHS2.java
@@ -84,7 +84,6 @@ public class MiniHS2 extends AbstractHiveService {
     hiveConf.setVar(ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST, getHost());
     hiveConf.setIntVar(ConfVars.HIVE_SERVER2_THRIFT_PORT, getBinaryPort());
     hiveConf.setIntVar(ConfVars.HIVE_SERVER2_THRIFT_HTTP_PORT, getHttpPort());
-    HiveMetaStore.HMSHandler.resetDefaultDBFlag();

     Path scratchDir = new Path(baseDfsDir, "scratch");
     fs.mkdirs(scratchDir);
@@ -209,4 +208,4 @@ public class MiniHS2 extends AbstractHiveService {
     } while (true);
   }

-}
\ No newline at end of file
+}
