*** oozie-5.0.0/core/src/main/java/org/apache/oozie/dependency/HCatURIHandler.java.old	2018-03-27 15:22:44.000000000 +0300
--- oozie-5.0.0/core/src/main/java/org/apache/oozie/dependency/HCatURIHandler.java	2018-07-30 10:33:54.117468097 +0300
***************
*** 47,53 ****
  import org.apache.oozie.service.URIHandlerService;
  import org.apache.oozie.util.HCatURI;
  import org.apache.oozie.util.XLog;
! import org.apache.hadoop.hive.thrift.DelegationTokenIdentifier;
  import org.apache.hadoop.io.Text;
  import org.apache.hadoop.security.token.Token;
  
--- 47,53 ----
  import org.apache.oozie.service.URIHandlerService;
  import org.apache.oozie.util.HCatURI;
  import org.apache.oozie.util.XLog;
! import org.apache.hadoop.hive.metastore.security.DelegationTokenIdentifier;
  import org.apache.hadoop.io.Text;
  import org.apache.hadoop.security.token.Token;
  
