diff --git a/core/src/main/java/org/apache/oozie/action/hadoop/JavaActionExecutor.java b/core/src/main/java/org/apache/oozie/action/hadoop/JavaActionExecutor.java
index 65996d9..be472af 100644
--- a/core/src/main/java/org/apache/oozie/action/hadoop/JavaActionExecutor.java
+++ b/core/src/main/java/org/apache/oozie/action/hadoop/JavaActionExecutor.java
@@ -44,7 +44,7 @@ import org.apache.hadoop.filecache.DistributedCache;
 import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
-import org.apache.hadoop.fs.permission.AccessControlException;
+import org.apache.hadoop.security.AccessControlException;
 import org.apache.oozie.hadoop.utils.HadoopShims;
 import org.apache.hadoop.io.Text;
 import org.apache.hadoop.mapred.JobClient;
diff --git a/core/src/main/java/org/apache/oozie/service/AuthorizationService.java b/core/src/main/java/org/apache/oozie/service/AuthorizationService.java
index 4e4ab17..149fbef 100644
--- a/core/src/main/java/org/apache/oozie/service/AuthorizationService.java
+++ b/core/src/main/java/org/apache/oozie/service/AuthorizationService.java
@@ -34,6 +34,7 @@ import java.util.Set;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.AccessControlException;
 import org.apache.oozie.BundleJobBean;
 import org.apache.oozie.CoordinatorJobBean;
 import org.apache.oozie.ErrorCode;
@@ -338,9 +339,7 @@ public class AuthorizationService implements Service {
                 }
                 fs.open(wfXml).close();
             }
-            // TODO change this when stopping support of 0.18 to the new
-            // Exception
-            catch (org.apache.hadoop.fs.permission.AccessControlException ex) {
+            catch (AccessControlException ex) {
                 incrCounter(INSTR_FAILED_AUTH_COUNTER, 1);
                 throw new AuthorizationException(ErrorCode.E0507, appPath, ex.getMessage(), ex);
             }
@@ -394,9 +393,7 @@ public class AuthorizationService implements Service {
                     }
                 }
             }
-            // TODO change this when stopping support of 0.18 to the new
-            // Exception
-            catch (org.apache.hadoop.fs.permission.AccessControlException ex) {
+            catch (AccessControlException ex) {
                 incrCounter(INSTR_FAILED_AUTH_COUNTER, 1);
                 throw new AuthorizationException(ErrorCode.E0507, appPath, ex.getMessage(), ex);
             }
