diff -ur zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/BaseSysTest.java zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/BaseSysTest.java
--- zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/BaseSysTest.java	2015-04-14 22:07:14.464000000 +0200
+++ zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/BaseSysTest.java	2015-04-14 22:07:18.728000000 +0200
@@ -37,6 +37,8 @@
 
 @Ignore("No tests in this class.")
 public class BaseSysTest extends TestCase {
+    private static final File testData = new File(
+            System.getProperty("test.data.dir", "build/test/data"));
     private static int fakeBasePort = 33222;
     private static String zkHostPort;
     protected String prefix = "/sysTest";
@@ -149,7 +151,7 @@
         }
         StringBuilder sb = new StringBuilder();
         for(int i = 0; i < count; i++) {
-            qpsDirs[i] = File.createTempFile("sysTest", ".tmp");
+            qpsDirs[i] = File.createTempFile("sysTest", ".tmp", testData);
             qpsDirs[i].delete();
             qpsDirs[i].mkdir();
             int port = fakeBasePort+10+i;
diff -ur zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java
--- zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java	2015-04-14 22:07:14.464000000 +0200
+++ zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java	2015-04-14 22:07:18.728000000 +0200
@@ -36,6 +36,9 @@
 
 class QuorumPeerInstance implements Instance {
     final private static Logger LOG = LoggerFactory.getLogger(QuorumPeerInstance.class);
+    private static final File testData = new File(
+        System.getProperty("test.data.dir", "build/test/data"));
+
     private static final int syncLimit = 3;
     private static final int initLimit = 3;
     private static final int tickTime = 2000;
@@ -55,7 +58,7 @@
 
     public QuorumPeerInstance() {
         try {
-            File tmpFile = File.createTempFile("test", ".dir");
+            File tmpFile = File.createTempFile("test", ".dir", testData);
             File tmpDir = tmpFile.getParentFile();
             tmpFile.delete();
             File zkDirs = new File(tmpDir, "zktmp.cfg");
diff -ur zookeeper-3.4.6/src_/java/test/org/apache/zookeeper/server/quorum/LearnerTest.java zookeeper-3.4.6/src/java/test/org/apache/zookeeper/server/quorum/LearnerTest.java
--- zookeeper-3.4.6/src_/java/test/org/apache/zookeeper/server/quorum/LearnerTest.java	2015-04-14 22:07:14.464000000 +0200
+++ zookeeper-3.4.6/src/java/test/org/apache/zookeeper/server/quorum/LearnerTest.java	2015-04-14 22:07:18.728000000 +0200
@@ -47,6 +47,9 @@
 import org.junit.Test;
 
 public class LearnerTest extends ZKTestCase {
+  private static final File testData = new File(
+    System.getProperty("test.data.dir", "build/test/data"));
+
 	class SimpleLearnerZooKeeperServer extends LearnerZooKeeperServer {
 		boolean startupCalled;
 		
@@ -84,7 +87,7 @@
 	}
 	@Test
 	public void syncTest() throws Exception {
-		File tmpFile = File.createTempFile("test", ".dir");
+		File tmpFile = File.createTempFile("test", ".dir", testData);
 		tmpFile.delete();
 		try {
 			FileTxnSnapLog ftsl = new FileTxnSnapLog(tmpFile, tmpFile);
diff -ur zookeeper-3.4.6/src_/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java zookeeper-3.4.6/src/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java
--- zookeeper-3.4.6/src_/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java	2015-04-14 22:07:14.464000000 +0200
+++ zookeeper-3.4.6/src/java/test/org/apache/zookeeper/server/quorum/Zab1_0Test.java	2015-04-14 22:07:18.728000000 +0200
@@ -80,6 +80,9 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(Zab1_0Test.class);
 
+    private static final File testData = new File(
+            System.getProperty("test.data.dir", "build/test/data"));
+
     private static final class LeadThread extends Thread {
         private final Leader leader;
 
@@ -149,7 +152,7 @@
     }
     @Test
     public void testLeaderInConnectingFollowers() throws Exception {    
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         Leader leader = null;
@@ -203,7 +206,7 @@
         
     @Test
     public void testLastAcceptedEpoch() throws Exception {    
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         Leader leader = null;
@@ -241,7 +244,7 @@
         
     @Test
     public void testLeaderInElectingFollowers() throws Exception {    
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         Leader leader = null;
@@ -348,7 +351,7 @@
         Socket pair[] = getSocketPair();
         Socket leaderSocket = pair[0];
         Socket followerSocket = pair[1];
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         LeadThread leadThread = null;
@@ -390,7 +393,7 @@
         Socket pair[] = getSocketPair();
         Socket leaderSocket = pair[0];
         Socket followerSocket = pair[1];
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         LeadThread leadThread = null;
@@ -457,7 +460,7 @@
     
     
     public void testFollowerConversation(FollowerConversation conversation) throws Exception {
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         Thread followerThread = null;
@@ -509,7 +512,7 @@
     }
 
     public void testObserverConversation(ObserverConversation conversation) throws Exception {
-        File tmpDir = File.createTempFile("test", "dir");
+        File tmpDir = File.createTempFile("test", "dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         Thread observerThread = null;
@@ -629,7 +632,7 @@
             @Override
             public void converseWithFollower(InputArchive ia, OutputArchive oa,
                     Follower f) throws Exception {
-                File tmpDir = File.createTempFile("test", "dir");
+                File tmpDir = File.createTempFile("test", "dir", testData);
                 tmpDir.delete();
                 tmpDir.mkdir();
                 File logDir = f.fzk.getTxnLogFactory().getDataDir().getParentFile();
@@ -757,7 +760,7 @@
             @Override
             public void converseWithFollower(InputArchive ia, OutputArchive oa,
                     Follower f) throws Exception {
-                File tmpDir = File.createTempFile("test", "dir");
+                File tmpDir = File.createTempFile("test", "dir", testData);
                 tmpDir.delete();
                 tmpDir.mkdir();
                 File logDir = f.fzk.getTxnLogFactory().getDataDir().getParentFile();
@@ -995,7 +998,7 @@
             @Override
             public void converseWithObserver(InputArchive ia, OutputArchive oa,
                     Observer o) throws Exception {
-                File tmpDir = File.createTempFile("test", "dir");
+                File tmpDir = File.createTempFile("test", "dir", testData);
                 tmpDir.delete();
                 tmpDir.mkdir();
                 File logDir = o.zk.getTxnLogFactory().getDataDir().getParentFile();
@@ -1413,7 +1416,7 @@
 
     @Test
     public void testInitialAcceptedCurrent() throws Exception {
-        File tmpDir = File.createTempFile("test", ".dir");
+        File tmpDir = File.createTempFile("test", ".dir", testData);
         tmpDir.delete();
         tmpDir.mkdir();
         try {
