diff -ur zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/Log4JSource.java zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/Log4JSource.java
--- zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/Log4JSource.java	2015-04-14 22:26:36.528000000 +0200
+++ zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/Log4JSource.java	2015-04-14 22:26:41.896000000 +0200
@@ -351,11 +351,21 @@
 		
 		LogIterator iter = s.iterator(starttime, endtime);
 		System.out.println(iter);
+		try {
+		  iter.close();
+		} catch (IOException ioe) {
+		  System.out.println(ioe.getMessage());
+		}
 	    }; };
 	    Thread t2 = new Thread() { public void run () { 
 		
 		LogIterator iter = s.iterator(starttime, endtime);
 		System.out.println(iter);
+		try {
+		  iter.close();
+		} catch (IOException ioe) {
+		  System.out.println(ioe.getMessage());
+		}
 	    }; };
 	    Thread t3 = new Thread() { public void run () { 
 		
diff -ur zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/RandomAccessFileReader.java zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/RandomAccessFileReader.java
--- zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/RandomAccessFileReader.java	2015-04-14 22:26:36.528000000 +0200
+++ zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/RandomAccessFileReader.java	2015-04-14 22:26:41.896000000 +0200
@@ -324,5 +324,6 @@
 	System.out.println(f.readLine());
 	f.seek(pos2);
 	System.out.println(f.readLine());
+	f.close();
     }
 };
diff -ur zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/NumEvents.java zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/NumEvents.java
--- zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/NumEvents.java	2015-04-14 22:26:36.528000000 +0200
+++ zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/NumEvents.java	2015-04-14 22:26:41.896000000 +0200
@@ -81,6 +81,7 @@
 	if (LOG.isDebugEnabled()) {
 	    LOG.debug("handle(start= " + starttime + ", end=" + endtime + ", numEntries=" + size +")");
 	}
+	iter.close();
 	return JSONValue.toJSONString(data);
     }
 }
diff -ur zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/StaticContent.java zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/StaticContent.java
--- zookeeper-3.4.6/src_/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/StaticContent.java	2015-04-14 22:26:36.528000000 +0200
+++ zookeeper-3.4.6/src/contrib/loggraph/src/java/org/apache/zookeeper/graph/servlets/StaticContent.java	2015-04-14 22:26:41.896000000 +0200
@@ -39,9 +39,12 @@
 	    response.setStatus(HttpServletResponse.SC_NOT_FOUND);
 	    return;
 	}
-	
-	while (resource.available() > 0) {
+	try {
+	  while (resource.available() > 0) {
 	    response.getWriter().write(resource.read());
+	  }
+	} finally {
+	  resource.close();
 	}
 	//        response.setContentType("text/plain;charset=utf-8");
         response.setStatus(HttpServletResponse.SC_OK);
diff -ur zookeeper-3.4.6/src_/contrib/rest/src/java/org/apache/zookeeper/server/jersey/cfg/RestCfg.java zookeeper-3.4.6/src/contrib/rest/src/java/org/apache/zookeeper/server/jersey/cfg/RestCfg.java
--- zookeeper-3.4.6/src_/contrib/rest/src/java/org/apache/zookeeper/server/jersey/cfg/RestCfg.java	2015-04-14 22:26:36.532000000 +0200
+++ zookeeper-3.4.6/src/contrib/rest/src/java/org/apache/zookeeper/server/jersey/cfg/RestCfg.java	2015-04-14 22:26:41.900000000 +0200
@@ -36,9 +36,13 @@
    }
 
    public RestCfg(InputStream io) throws IOException {
+     try {
        cfg.load(io);
        extractEndpoints();
        extractCredentials();
+     } finally {
+       io.close();
+     }
    }
 
    private void extractCredentials() {
diff -ur zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/GenerateLoad.java zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/GenerateLoad.java
--- zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/GenerateLoad.java	2015-04-14 22:26:36.524000000 +0200
+++ zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/GenerateLoad.java	2015-04-14 22:26:41.900000000 +0200
@@ -699,12 +699,16 @@
         s.getOutputStream().write("stat".getBytes());
         BufferedReader br = new BufferedReader(new InputStreamReader(s.getInputStream()));
         String line;
-        while((line = br.readLine()) != null) {
+        try {
+          while((line = br.readLine()) != null) {
             if (line.startsWith("Mode: ")) {
-                return line.substring(6);
+              return line.substring(6);
             }
+          }
+          return "unknown";
+        } finally {
+          s.close();
         }
-        return "unknown";
     }
 
     private static void doUsage() {
diff -ur zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java
--- zookeeper-3.4.6/src_/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java	2015-04-14 22:26:36.524000000 +0200
+++ zookeeper-3.4.6/src/java/systest/org/apache/zookeeper/test/system/QuorumPeerInstance.java	2015-04-14 22:26:41.900000000 +0200
@@ -67,7 +67,12 @@
             Properties p;
             if (zkDirs.exists()) {
                 p = new Properties();
-                p.load(new FileInputStream(zkDirs));
+                FileInputStream input = new FileInputStream(zkDirs);
+                try {
+                  p.load(input);
+                } finally {
+                  input.close();
+                }
             } else {
                 p = System.getProperties();
             }
