diff --git a/src/java/main/org/apache/zookeeper/ZooKeeperMain.java b/src/java/main/org/apache/zookeeper/ZooKeeperMain.java
index b1b4c27..39fc465 100644
--- a/src/java/main/org/apache/zookeeper/ZooKeeperMain.java
+++ b/src/java/main/org/apache/zookeeper/ZooKeeperMain.java
@@ -349,6 +349,9 @@ public class ZooKeeperMain {
                     executeLine(line);
                 }
             }
+        } else {
+            // Command line args non-null.  Run what was passed.
+            processCmd(cl);
         }
     }
 
