diff -uraN apache-apex-core-3.6.0-orig/engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java apache-apex-core-3.6.0/engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java
--- apache-apex-core-3.6.0-orig/engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java	2017-04-28 20:55:41.000000000 +0200
+++ apache-apex-core-3.6.0/engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java	2017-05-27 23:44:35.028907503 +0200
@@ -584,7 +584,7 @@
     if (UserGroupInformation.isSecurityEnabled()) {
       long hdfsTokenMaxLifeTime = conf.getLong(StramClientUtils.DT_HDFS_TOKEN_MAX_LIFE_TIME, conf.getLong(StramClientUtils.HDFS_TOKEN_MAX_LIFE_TIME, StramClientUtils.DELEGATION_TOKEN_MAX_LIFETIME_DEFAULT));
       dag.setAttribute(LogicalPlan.HDFS_TOKEN_LIFE_TIME, hdfsTokenMaxLifeTime);
-      long rmTokenMaxLifeTime = conf.getLong(StramClientUtils.DT_RM_TOKEN_MAX_LIFE_TIME, conf.getLong(YarnConfiguration.DELEGATION_TOKEN_MAX_LIFETIME_KEY, YarnConfiguration.DELEGATION_TOKEN_MAX_LIFETIME_DEFAULT));
+      long rmTokenMaxLifeTime = conf.getLong(StramClientUtils.DT_RM_TOKEN_MAX_LIFE_TIME, conf.getLong(YarnConfiguration.RM_DELEGATION_TOKEN_MAX_LIFETIME_KEY, YarnConfiguration.RM_DELEGATION_TOKEN_MAX_LIFETIME_DEFAULT));
       dag.setAttribute(LogicalPlan.RM_TOKEN_LIFE_TIME, rmTokenMaxLifeTime);
       setTokenRefreshCredentials(dag, conf);
     }
diff -uraN apache-apex-core-3.6.0-orig/engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java apache-apex-core-3.6.0/engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java
--- apache-apex-core-3.6.0-orig/engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java	2017-04-28 20:55:41.000000000 +0200
+++ apache-apex-core-3.6.0/engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java	2017-05-27 23:44:20.360926755 +0200
@@ -162,7 +162,7 @@
    */
   public static Attribute<Boolean> FAST_PUBLISHER_SUBSCRIBER = new Attribute<>(false);
   public static Attribute<Long> HDFS_TOKEN_LIFE_TIME = new Attribute<>(604800000L);
-  public static Attribute<Long> RM_TOKEN_LIFE_TIME = new Attribute<>(YarnConfiguration.DELEGATION_TOKEN_MAX_LIFETIME_DEFAULT);
+  public static Attribute<Long> RM_TOKEN_LIFE_TIME = new Attribute<>(YarnConfiguration.RM_DELEGATION_TOKEN_MAX_LIFETIME_DEFAULT);
   public static Attribute<String> PRINCIPAL = new Attribute<>(null, StringCodec.String2String.getInstance());
   public static Attribute<String> KEY_TAB_FILE = new Attribute<>((String)null, StringCodec.String2String.getInstance());
   public static Attribute<Double> TOKEN_REFRESH_ANTICIPATORY_FACTOR = new Attribute<>(0.7);
