diff --git hadoop-common-project/hadoop-common/src/main/bin/hadoop-config.sh hadoop-common-project/hadoop-common/src/main/bin/hadoop-config.sh
index 5f248fd..8bf9aec 100644
--- hadoop-common-project/hadoop-common/src/main/bin/hadoop-config.sh
+++ hadoop-common-project/hadoop-common/src/main/bin/hadoop-config.sh
@@ -27,7 +27,9 @@ this="$common_bin/$script"
 
 # the root of the Hadoop installation
 # See HADOOP-6255 for directory structure layout
-export HADOOP_PREFIX=`dirname "$this"`/..
+HADOOP_DEFAULT_PREFIX=`dirname "$this"`/..
+HADOOP_PREFIX=${HADOOP_PREFIX:-$HADOOP_DEFAULT_PREFIX}
+export HADOOP_PREFIX
 
 #check to see if the conf dir is given as an optional argument
 if [ $# -gt 1 ]
diff --git hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-datanode hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-datanode
index 8179587..658f6d7 100644
--- hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-datanode
+++ hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-datanode
@@ -75,7 +75,7 @@ check_privsep_dir() {
 }
 
 export PATH="${PATH:+$PATH:}/usr/sbin:/usr/bin"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 case "$1" in
   start)
diff --git hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-jobtracker hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-jobtracker
index c483d76..8958da3 100644
--- hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-jobtracker
+++ hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-jobtracker
@@ -67,7 +67,7 @@ check_privsep_dir() {
 }
 
 export PATH="${PATH:+$PATH:}/usr/sbin:/usr/bin"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 case "$1" in
   start)
diff --git hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-namenode hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-namenode
index e260876..6e5485c 100644
--- hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-namenode
+++ hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-namenode
@@ -71,7 +71,7 @@ format() {
 }
 
 export PATH="${PATH:+$PATH:}/usr/sbin:/usr/bin"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 case "$1" in
   start)
diff --git hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-tasktracker hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-tasktracker
index 983030b..ca36cd6 100644
--- hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-tasktracker
+++ hadoop-common-project/hadoop-common/src/main/packages/deb/init.d/hadoop-tasktracker
@@ -67,7 +67,7 @@ check_privsep_dir() {
 }
 
 export PATH="${PATH:+$PATH:}/usr/sbin:/usr/bin"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 case "$1" in
   start)
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh
index db1f5b9..de47ce2 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh
@@ -19,7 +19,9 @@ bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
 this="${BASH_SOURCE-$0}"
-export HADOOP_PREFIX=`dirname "$this"`/..
+HADOOP_DEFAULT_PREFIX=`dirname "$this"`/..
+HADOOP_PREFIX=${HADOOP_PREFIX:-$HADOOP_DEFAULT_PREFIX}
+export HADOOP_PREFIX
 
 usage() {
   echo "
diff --git hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-jobtracker hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-jobtracker
index 5c2272b..9e85c66 100644
--- hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-jobtracker
+++ hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-jobtracker
@@ -27,7 +27,7 @@ source /etc/default/hadoop-env.sh
 RETVAL=0
 PIDFILE="${HADOOP_PID_DIR}/hadoop-mapred-jobtracker.pid"
 desc="Hadoop jobtracker daemon"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 start() {
   echo -n $"Starting $desc (hadoop-jobtracker): "
diff --git hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-namenode hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-namenode
index 4e7f74b..3067f03 100644
--- hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-namenode
+++ hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-namenode
@@ -27,7 +27,7 @@ source /etc/default/hadoop-env.sh
 RETVAL=0
 PIDFILE="${HADOOP_PID_DIR}/hadoop-hdfs-namenode.pid"
 desc="Hadoop namenode daemon"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 start() {
   echo -n $"Starting $desc (hadoop-namenode): "
diff --git hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-tasktracker hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-tasktracker
index 2d06c5a..9a57f56 100644
--- hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-tasktracker
+++ hadoop-common-project/hadoop-common/src/main/packages/rpm/init.d/hadoop-tasktracker
@@ -27,7 +27,7 @@ source /etc/default/hadoop-env.sh
 RETVAL=0
 PIDFILE="${HADOOP_PID_DIR}/hadoop-mapred-tasktracker.pid"
 desc="Hadoop tasktracker daemon"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 start() {
   echo -n $"Starting $desc (hadoop-tasktracker): "
