diff --git hadoop-common-project/hadoop-common/src/main/bin/hadoop hadoop-common-project/hadoop-common/src/main/bin/hadoop
index a3bed69..9e92b5c 100755
--- hadoop-common-project/hadoop-common/src/main/bin/hadoop
+++ hadoop-common-project/hadoop-common/src/main/bin/hadoop
@@ -21,7 +21,9 @@ bin=`which $0`
 bin=`dirname ${bin}`
 bin=`cd "$bin"; pwd`
  
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 function print_usage(){
   echo "Usage: hadoop [--config confdir] COMMAND"
diff --git hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemon.sh hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemon.sh
index 01aaad4..a843c93 100755
--- hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemon.sh
+++ hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemon.sh
@@ -39,7 +39,9 @@ fi
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # get arguments
 
diff --git hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemons.sh hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemons.sh
index 08c7e23..181d7ac 100755
--- hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemons.sh
+++ hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemons.sh
@@ -29,6 +29,8 @@ fi
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 exec "$bin/slaves.sh" --config $HADOOP_CONF_DIR cd "$HADOOP_PREFIX" \; "$bin/hadoop-daemon.sh" --config $HADOOP_CONF_DIR "$@"
diff --git hadoop-common-project/hadoop-common/src/main/bin/rcc hadoop-common-project/hadoop-common/src/main/bin/rcc
index ebeebd0..5f75b7c 100755
--- hadoop-common-project/hadoop-common/src/main/bin/rcc
+++ hadoop-common-project/hadoop-common/src/main/bin/rcc
@@ -30,7 +30,9 @@
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 if [ -f "${HADOOP_CONF_DIR}/hadoop-env.sh" ]; then
   . "${HADOOP_CONF_DIR}/hadoop-env.sh"
diff --git hadoop-common-project/hadoop-common/src/main/bin/slaves.sh hadoop-common-project/hadoop-common/src/main/bin/slaves.sh
index 153f441..016392f 100755
--- hadoop-common-project/hadoop-common/src/main/bin/slaves.sh
+++ hadoop-common-project/hadoop-common/src/main/bin/slaves.sh
@@ -38,7 +38,9 @@ fi
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 if [ -f "${HADOOP_CONF_DIR}/hadoop-env.sh" ]; then
   . "${HADOOP_CONF_DIR}/hadoop-env.sh"
diff --git hadoop-common-project/hadoop-common/src/main/bin/start-all.sh hadoop-common-project/hadoop-common/src/main/bin/start-all.sh
index 57fb3d6..f91d9af 100755
--- hadoop-common-project/hadoop-common/src/main/bin/start-all.sh
+++ hadoop-common-project/hadoop-common/src/main/bin/start-all.sh
@@ -23,7 +23,9 @@ echo "This script is Deprecated. Instead use start-dfs.sh and start-mapred.sh"
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # start hdfs daemons if hdfs is present
 if [ -f "${HADOOP_HDFS_HOME}"/bin/start-dfs.sh ]; then
diff --git hadoop-common-project/hadoop-common/src/main/bin/stop-all.sh hadoop-common-project/hadoop-common/src/main/bin/stop-all.sh
index a2b5ddb..7d8bd59 100755
--- hadoop-common-project/hadoop-common/src/main/bin/stop-all.sh
+++ hadoop-common-project/hadoop-common/src/main/bin/stop-all.sh
@@ -23,7 +23,9 @@ echo "This script is Deprecated. Instead use stop-dfs.sh and stop-mapred.sh"
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # stop hdfs daemons if hdfs is present
 if [ -f "${HADOOP_HDFS_HOME}"/bin/stop-dfs.sh ]; then
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-create-user.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-create-user.sh
index a5d4304..ad8ab35 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-create-user.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-create-user.sh
@@ -24,7 +24,9 @@ if [ "$HADOOP_HOME" != "" ]; then
   echo
 fi
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 usage() {
   echo "
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-applications.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-applications.sh
index 521c159..092e50d 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-applications.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-applications.sh
@@ -19,7 +19,9 @@ bin=$(cd -P -- "$(dirname -- "$this")" && pwd -P)
 script="$(basename -- "$this")"
 this="$bin/$script"
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 usage() {
   echo "
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh
index db1f5b9..fa3cc6a 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-conf.sh
@@ -502,7 +502,10 @@ if [ "${AUTOSETUP}" == "1" -o "${AUTOSETUP}" == "y" ]; then
   fi
   chmod 755 -R ${HADOOP_PREFIX}/sbin/*hadoop*
   chmod 755 -R ${HADOOP_PREFIX}/bin/hadoop
-  chmod 755 -R ${HADOOP_PREFIX}/libexec/hadoop-config.sh
+
+  HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-${HADOOP_PREFIX}/libexec}
+  chmod 755 -R ${HADOOP_LIBEXEC_DIR}/hadoop-config.sh
+
   mkdir -p /home/${HADOOP_MR_USER}
   chown ${HADOOP_MR_USER}:${HADOOP_GROUP} /home/${HADOOP_MR_USER}
   HDFS_DIR=`echo ${HADOOP_HDFS_DIR} | sed -e 's/,/ /g'`
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-hdfs.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-hdfs.sh
index fc4a732..d2e8f1f 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-hdfs.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-hdfs.sh
@@ -18,7 +18,9 @@
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 usage() {
   echo "
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-single-node.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-single-node.sh
index 5b47e44..845f815 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-single-node.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-setup-single-node.sh
@@ -25,7 +25,9 @@ if [ "$HADOOP_HOME" != "" ]; then
   echo
 fi
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 usage() {
   echo "
diff --git hadoop-common-project/hadoop-common/src/main/packages/hadoop-validate-setup.sh hadoop-common-project/hadoop-common/src/main/packages/hadoop-validate-setup.sh
index 5d3aa14..35d8cb8 100644
--- hadoop-common-project/hadoop-common/src/main/packages/hadoop-validate-setup.sh
+++ hadoop-common-project/hadoop-common/src/main/packages/hadoop-validate-setup.sh
@@ -31,7 +31,9 @@
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 usage() {
   echo "
