diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/distribute-exclude.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/distribute-exclude.sh
index cc538f7..66fc14a 100644
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/distribute-exclude.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/distribute-exclude.sh
@@ -36,7 +36,9 @@
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin/../libexec/hdfs-config.sh"
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 if [ "$1" = '' ] ; then
   "Error: please specify local exclude file as a first argument"
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs
index d9b8f61..314fac8 100755
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs
@@ -19,7 +19,9 @@ bin=`which $0`
 bin=`dirname ${bin}`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 function print_usage(){
   echo "Usage: hdfs [--config confdir] COMMAND"
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs-config.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs-config.sh
index 48aa20c..09eec6e 100644
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs-config.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/hdfs-config.sh
@@ -24,8 +24,10 @@ bin=`cd "$bin"; pwd`
 
 export HADOOP_PREFIX="${HADOOP_PREFIX:-$bin/..}"
 
-if [ -e "$bin/../libexec/hadoop-config.sh" ]; then
-  . $bin/../libexec/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+if [ -e "${HADOOP_LIBEXEC_DIR}/hadoop-config.sh" ]; then
+  . ${HADOOP_LIBEXEC_DIR}/hadoop-config.sh
 elif [ -e "${HADOOP_COMMON_HOME}/libexec/hadoop-config.sh" ]; then
   . "$HADOOP_COMMON_HOME"/libexec/hadoop-config.sh
 elif [ -e "${HADOOP_HOME}/libexec/hadoop-config.sh" ]; then
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/refresh-namenodes.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/refresh-namenodes.sh
index 2092764..d3f6759 100644
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/refresh-namenodes.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/refresh-namenodes.sh
@@ -23,7 +23,9 @@
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin/../libexec/hdfs-config.sh"
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 namenodes=$("$HADOOP_PREFIX/bin/hdfs" getconf -nnRpcAddresses)
 if [ "$?" != '0' ] ; then errorFlag='1' ; 
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-balancer.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-balancer.sh
index b6b3aa7..24c6227 100755
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-balancer.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-balancer.sh
@@ -18,7 +18,9 @@
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 # Start balancer daemon.
 
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-dfs.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-dfs.sh
index d6ed5f9..d6d03f7 100755
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-dfs.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-dfs.sh
@@ -25,7 +25,9 @@ usage="Usage: start-dfs.sh [-upgrade|-rollback]"
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 # get arguments
 if [ $# -ge 1 ]; then
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-secure-dns.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-secure-dns.sh
index bcb9b8f..7ddf687 100644
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-secure-dns.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/start-secure-dns.sh
@@ -22,7 +22,9 @@ usage="Usage (run as root in order to start secure datanodes): start-secure-dns.
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 if [ "$EUID" -eq 0 ] && [ -n "$HADOOP_SECURE_DN_USER" ]; then
   "$HADOOP_PREFIX"/sbin/hadoop-daemons.sh --config $HADOOP_CONF_DIR --script "$bin"/hdfs start datanode $dataStartOpt
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-balancer.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-balancer.sh
index 7edd0bd..5026c8c 100755
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-balancer.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-balancer.sh
@@ -18,7 +18,9 @@
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 # Stop balancer daemon.
 # Run this on the machine where the balancer is running
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-dfs.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-dfs.sh
index e64a5ea..11788e2 100755
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-dfs.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-dfs.sh
@@ -18,7 +18,9 @@
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 #---------------------------------------------------------
 # namenodes
diff --git hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-secure-dns.sh hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-secure-dns.sh
index a0ae172..fdd47c3 100644
--- hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-secure-dns.sh
+++ hadoop-hdfs-project/hadoop-hdfs/src/main/bin/stop-secure-dns.sh
@@ -22,7 +22,9 @@ usage="Usage (run as root in order to stop secure datanodes): stop-secure-dns.sh
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/../libexec/hdfs-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hdfs-config.sh
 
 if [ "$EUID" -eq 0 ] && [ -n "$HADOOP_SECURE_DN_USER" ]; then
   "$HADOOP_PREFIX"/sbin/hadoop-daemons.sh --config $HADOOP_CONF_DIR --script "$bin"/hdfs stop datanode
