diff --git hadoop-mapreduce-project/src/examples/python/compile hadoop-mapreduce-project/src/examples/python/compile
index e202a1c..32c80ed 100644
--- hadoop-mapreduce-project/src/examples/python/compile
+++ hadoop-mapreduce-project/src/examples/python/compile
@@ -12,7 +12,7 @@
 #   limitations under the License.
 
 
-export HADOOP_PREFIX=../../..
+export HADOOP_PREFIX=${HADOOP_PREFIX:-../../..}
 
 export CLASSPATH="$HADOOP_PREFIX/build/classes"
 
diff --git hadoop-mapreduce-project/src/examples/python/pyAbacus/compile hadoop-mapreduce-project/src/examples/python/pyAbacus/compile
index c06573b..641d5ff 100644
--- hadoop-mapreduce-project/src/examples/python/pyAbacus/compile
+++ hadoop-mapreduce-project/src/examples/python/pyAbacus/compile
@@ -12,7 +12,7 @@
 #   limitations under the License.
 
 
-export HADOOP_PREFIX=../../../../..
+export HADOOP_PREFIX=${HADOOP_PREFIX:-../../../../..}
 
 export CLASSPATH="$HADOOP_PREFIX/build/classes"
 export CLASSPATH=${CLASSPATH}:"$HADOOP_PREFIX/build/contrib/abacus/classes"
diff --git hadoop-mapreduce-project/src/packages/deb/init.d/hadoop-historyserver hadoop-mapreduce-project/src/packages/deb/init.d/hadoop-historyserver
index 4421f55..6334729 100644
--- hadoop-mapreduce-project/src/packages/deb/init.d/hadoop-historyserver
+++ hadoop-mapreduce-project/src/packages/deb/init.d/hadoop-historyserver
@@ -67,7 +67,7 @@ check_privsep_dir() {
 }
 
 export PATH="${PATH:+$PATH:}/usr/sbin:/usr/bin"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 case "$1" in
   start)
diff --git hadoop-mapreduce-project/src/packages/rpm/init.d/hadoop-historyserver hadoop-mapreduce-project/src/packages/rpm/init.d/hadoop-historyserver
index 71d1658..5677593 100644
--- hadoop-mapreduce-project/src/packages/rpm/init.d/hadoop-historyserver
+++ hadoop-mapreduce-project/src/packages/rpm/init.d/hadoop-historyserver
@@ -27,7 +27,7 @@ source /etc/default/hadoop-env.sh
 RETVAL=0
 PIDFILE="${HADOOP_PID_DIR}/hadoop-mapred-historyserver.pid"
 desc="Hadoop historyserver daemon"
-export HADOOP_PREFIX="/usr"
+export HADOOP_PREFIX=${HADOOP_PREFIX:-/usr}
 
 start() {
   echo -n $"Starting $desc (hadoop-historyserver): "
