diff --git hadoop-mapreduce-project/bin/mapred hadoop-mapreduce-project/bin/mapred
index e5e9efb..de82a03 100755
--- hadoop-mapreduce-project/bin/mapred
+++ hadoop-mapreduce-project/bin/mapred
@@ -19,8 +19,10 @@ bin=`which $0`
 bin=`dirname ${bin}`
 bin=`cd "$bin"; pwd`
 
-if [ -e $bin/../libexec/mapred-config.sh ]; then
-  . $bin/../libexec/mapred-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+if [ -e ${HADOOP_LIBEXEC_DIR}/mapred-config.sh ]; then
+  . ${HADOOP_LIBEXEC_DIR}/mapred-config.sh
 else
   . "$bin/mapred-config.sh"
 fi
diff --git hadoop-mapreduce-project/bin/mapred-config.sh hadoop-mapreduce-project/bin/mapred-config.sh
index c79f4fb..d1eb627 100644
--- hadoop-mapreduce-project/bin/mapred-config.sh
+++ hadoop-mapreduce-project/bin/mapred-config.sh
@@ -22,8 +22,10 @@ bin=`which "$0"`
 bin=`dirname "${bin}"`
 bin=`cd "$bin"; pwd`
 
-if [ -e "$bin/../libexec/hadoop-config.sh" ]; then
-  . "$bin/../libexec/hadoop-config.sh"
+DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+if [ -e "${HADOOP_LIBEXEC_DIR}/hadoop-config.sh" ]; then
+  . "${HADOOP_LIBEXEC_DIR}/hadoop-config.sh"
 elif [ -e "${HADOOP_COMMON_HOME}/libexec/hadoop-config.sh" ]; then
   . "$HADOOP_COMMON_HOME"/libexec/hadoop-config.sh
 elif [ -e "${HADOOP_COMMON_HOME}/bin/hadoop-config.sh" ]; then
diff --git hadoop-mapreduce-project/hadoop-yarn/bin/slaves.sh hadoop-mapreduce-project/hadoop-yarn/bin/slaves.sh
index e32a451..ee83477 100644
--- hadoop-mapreduce-project/hadoop-yarn/bin/slaves.sh
+++ hadoop-mapreduce-project/hadoop-yarn/bin/slaves.sh
@@ -38,7 +38,9 @@ fi
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/yarn-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/yarn-config.sh
 
 # If the slaves file is specified in the command line,
 # then it takes precedence over the definition in 
diff --git hadoop-mapreduce-project/hadoop-yarn/bin/start-all.sh hadoop-mapreduce-project/hadoop-yarn/bin/start-all.sh
index e1a798f..fa4fcf3 100644
--- hadoop-mapreduce-project/hadoop-yarn/bin/start-all.sh
+++ hadoop-mapreduce-project/hadoop-yarn/bin/start-all.sh
@@ -23,7 +23,10 @@ echo "starting yarn daemons"
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/yarn-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/yarn-config.sh
+
 # start resourceManager
 "$bin"/yarn-daemon.sh --config $YARN_CONF_DIR  start resourcemanager
 # start nodeManager
diff --git hadoop-mapreduce-project/hadoop-yarn/bin/stop-all.sh hadoop-mapreduce-project/hadoop-yarn/bin/stop-all.sh
index 850af3e..546b67f 100644
--- hadoop-mapreduce-project/hadoop-yarn/bin/stop-all.sh
+++ hadoop-mapreduce-project/hadoop-yarn/bin/stop-all.sh
@@ -23,7 +23,10 @@ echo "stopping yarn daemons"
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/yarn-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/yarn-config.sh
+
 # stop resourceManager
 "$bin"/yarn-daemon.sh --config $YARN_CONF_DIR  stop resourcemanager
 # stop nodeManager
diff --git hadoop-mapreduce-project/hadoop-yarn/bin/yarn hadoop-mapreduce-project/hadoop-yarn/bin/yarn
index 059bf10..b11c94b 100644
--- hadoop-mapreduce-project/hadoop-yarn/bin/yarn
+++ hadoop-mapreduce-project/hadoop-yarn/bin/yarn
@@ -44,7 +44,9 @@
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/yarn-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/yarn-config.sh
 
 cygwin=false
 case "`uname`" in
diff --git hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemon.sh hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemon.sh
index c3d1724..39bb76c 100644
--- hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemon.sh
+++ hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemon.sh
@@ -39,7 +39,9 @@ fi
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/yarn-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/yarn-config.sh
 
 # get arguments
 startStop=$1
diff --git hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemons.sh hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemons.sh
index 4f89a68..e34e4ca 100644
--- hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemons.sh
+++ hadoop-mapreduce-project/hadoop-yarn/bin/yarn-daemons.sh
@@ -30,7 +30,9 @@ fi
 bin=`dirname "${BASH_SOURCE-$0}"`
 bin=`cd "$bin"; pwd`
 
-. $bin/yarn-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/yarn-config.sh
 
 exec "$bin/slaves.sh" --config $YARN_CONF_DIR cd "$YARN_HOME" \; "$bin/yarn-daemon.sh" --config $YARN_CONF_DIR "$@"
 
diff --git hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode-remote.sh hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode-remote.sh
index c0aefd3..3e9405e 100644
--- hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode-remote.sh
+++ hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode-remote.sh
@@ -24,7 +24,9 @@ params=$#
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # get arguments
 if [ $# -ge 1 ]; then
diff --git hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode.sh hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode.sh
index 846aa80..d193bb1 100644
--- hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode.sh
+++ hadoop-mapreduce-project/src/contrib/raid/bin/start-raidnode.sh
@@ -25,7 +25,9 @@ params=$#
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # get arguments
 if [ $# -ge 1 ]; then
diff --git hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode-remote.sh hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode-remote.sh
index ed870ca..d185629 100644
--- hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode-remote.sh
+++ hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode-remote.sh
@@ -24,7 +24,9 @@ params=$#
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # get arguments
 if [ $# -ge 1 ]; then
diff --git hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode.sh hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode.sh
index d207e5a..02a61a8 100644
--- hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode.sh
+++ hadoop-mapreduce-project/src/contrib/raid/bin/stop-raidnode.sh
@@ -25,7 +25,9 @@ params=$#
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
-. "$bin"/hadoop-config.sh
+DEFAULT_LIBEXEC_DIR="$bin"
+HADOOP_LIBEXEC_DIR=${HADOOP_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
+. $HADOOP_LIBEXEC_DIR/hadoop-config.sh
 
 # get arguments
 if [ $# -ge 1 ]; then
