diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/pom.xml b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/pom.xml
index 2d0c4fa705c..9777b14595f 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/pom.xml
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/pom.xml
@@ -30,7 +30,6 @@
     <!-- Basedir eeded for generating FindBugs warnings using parent pom -->
     <yarn.basedir>${project.parent.parent.basedir}</yarn.basedir>
     <container-executor.conf.dir>../etc/hadoop</container-executor.conf.dir>
-    <extra.libhadoop.rpath>../lib/native</extra.libhadoop.rpath>
     <container-executor.additional_cflags></container-executor.additional_cflags>
   </properties>
 
@@ -200,7 +199,6 @@
                   <source>${basedir}/src</source>
                   <vars>
                     <HADOOP_CONF_DIR>${container-executor.conf.dir}</HADOOP_CONF_DIR>
-                    <EXTRA_LIBHADOOP_RPATH>${extra.libhadoop.rpath}</EXTRA_LIBHADOOP_RPATH>
                     <JVM_ARCH_DATA_MODEL>${sun.arch.data.model}</JVM_ARCH_DATA_MODEL>
                   </vars>
                   <env>
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/CMakeLists.txt b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/CMakeLists.txt
index e2cfbc52df2..677429bb99c 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/CMakeLists.txt
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/CMakeLists.txt
@@ -25,8 +25,6 @@ set(GTEST_SRC_DIR ${CMAKE_SOURCE_DIR}/../../../../../hadoop-common-project/hadoo
 
 set(HADOOP_COMMON_SEC_PATH ${HADOOP_COMMON_PATH}/src/main/native/src/org/apache/hadoop/security)
 
-set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-
 # determine if container-executor.conf.dir is an absolute
 # path in case the OS we're compiling on doesn't have
 # a hook in get_executable. We'll use this define
@@ -159,18 +157,6 @@ add_executable(container-executor
     main/native/container-executor/impl/main.c
 )
 
-# By embedding '$ORIGIN' into the RPATH of container-executor, dlopen will look in
-# the directory containing container-executor. However, $ORIGIN is not supported by
-# all operating systems.
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux|SunOS")
-    set(RPATH "\$ORIGIN/")
-    if(EXTRA_LIBHADOOP_RPATH)
-        set(RPATH "${RPATH}:${EXTRA_LIBHADOOP_RPATH}/")
-    endif()
-    message("RPATH SET AS ${RPATH}.")
-    set_target_properties(container-executor PROPERTIES INSTALL_RPATH "${RPATH}")
-endif()
-
 target_link_libraries(container-executor
     container
     crypto
@@ -182,19 +168,6 @@ output_directory(container-executor target/usr/local/bin)
 add_executable(test-container-executor
     main/native/container-executor/test/test-container-executor.c
 )
-
-# By embedding '$ORIGIN' into the RPATH of test-container-executor, dlopen will look in
-# the directory containing test-container-executor. However, $ORIGIN is not supported by
-# all operating systems.
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux|SunOS")
-    set(RPATH "\$ORIGIN/")
-    if(EXTRA_LIBHADOOP_RPATH)
-        set(RPATH "${RPATH}:${EXTRA_LIBHADOOP_RPATH}/")
-    endif()
-    message("RPATH SET AS ${RPATH}.")
-    set_target_properties(test-container-executor PROPERTIES INSTALL_RPATH "${RPATH}")
-endif()
-
 target_link_libraries(test-container-executor
     container
     ${EXTRA_LIBS}
