diff --git a/hadoop-common-project/hadoop-kms/src/main/libexec/shellprofile.d/hadoop-kms.sh b/hadoop-common-project/hadoop-kms/src/main/libexec/shellprofile.d/hadoop-kms.sh
index b54bf811d6b..bb860ce2d93 100755
--- a/hadoop-common-project/hadoop-kms/src/main/libexec/shellprofile.d/hadoop-kms.sh
+++ b/hadoop-common-project/hadoop-kms/src/main/libexec/shellprofile.d/hadoop-kms.sh
@@ -51,9 +51,4 @@ function hadoop_subcommand_kms
     "-Dkms.log.dir=${HADOOP_LOG_DIR}"
   hadoop_add_param HADOOP_OPTS "-Dlog4j.configuration=" \
     "-Dlog4j.configuration=file:${HADOOP_CONF_DIR}/kms-log4j.properties"
-
-  if [[ "${HADOOP_DAEMON_MODE}" == "default" ]] ||
-     [[ "${HADOOP_DAEMON_MODE}" == "start" ]]; then
-    hadoop_mkdir "${KMS_TEMP:-${HADOOP_HOME}/temp}"
-  fi
 }
