diff --git a/tez-dag/src/main/java/org/apache/tez/state/StateMachineTez.java b/tez-dag/src/main/java/org/apache/tez/state/StateMachineTez.java
index 3be771892..cbb838d7f 100644
--- a/tez-dag/src/main/java/org/apache/tez/state/StateMachineTez.java
+++ b/tez-dag/src/main/java/org/apache/tez/state/StateMachineTez.java
@@ -51,6 +51,11 @@ public class StateMachineTez<STATE extends Enum<STATE>, EVENTTYPE extends Enum<E
     return realStatemachine.getCurrentState();
   }
 
+  @Override
+  public STATE getPreviousState() {
+    return realStatemachine.getPreviousState();
+  }
+
   @SuppressWarnings("unchecked")
   @Override
   public STATE doTransition(EVENTTYPE eventType, EVENT event) throws
