diff --git a/tez-plugins/tez-aux-services/findbugs-exclude.xml b/tez-plugins/tez-aux-services/findbugs-exclude.xml
index 5b11308f6..adfd7041a 100644
--- a/tez-plugins/tez-aux-services/findbugs-exclude.xml
+++ b/tez-plugins/tez-aux-services/findbugs-exclude.xml
@@ -12,5 +12,10 @@
   limitations under the License. See accompanying LICENSE file.
 -->
 <FindBugsFilter>
-
+  <!-- TEZ-4298 -->
+  <Match>
+    <Class name="org.apache.tez.auxservices.ShuffleHandler"/>
+    <Method name="recordJobShuffleInfo"/>
+    <Bug pattern="RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT"/>
+  </Match>
 </FindBugsFilter>
diff --git a/tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java b/tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java
index e73805a1f..fb28a0f4b 100644
--- a/tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java
+++ b/tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java
@@ -153,7 +153,6 @@ import com.google.common.cache.LoadingCache;
 import com.google.common.cache.RemovalListener;
 import com.google.common.cache.RemovalNotification;
 import com.google.common.cache.Weigher;
-import com.google.protobuf.ByteString;
 
 public class ShuffleHandler extends AuxiliaryService {
 
@@ -796,9 +795,10 @@ public class ShuffleHandler extends AuxiliaryService {
   private void recordJobShuffleInfo(JobID jobId, String user,
       Token<JobTokenIdentifier> jobToken) throws IOException {
     if (stateDb != null) {
+      // Discover type instead of assuming ByteString to allow for shading.
       TokenProto tokenProto = TokenProto.newBuilder()
-          .setIdentifier(ByteString.copyFrom(jobToken.getIdentifier()))
-          .setPassword(ByteString.copyFrom(jobToken.getPassword()))
+          .setIdentifier(TokenProto.getDefaultInstance().getIdentifier().copyFrom(jobToken.getIdentifier()))
+          .setPassword(TokenProto.getDefaultInstance().getPassword().copyFrom(jobToken.getPassword()))
           .setKind(jobToken.getKind().toString())
           .setService(jobToken.getService().toString())
           .build();
