--- kibana-5.4.1/tasks/config/platforms.js     2017-05-27 12:53:15.000000000 +0000
+++ kibana-5.4.1/tasks/config/platforms.js     2021-07-14 21:37:50.423345645 +0000
@@ -12,6 +12,8 @@
     'darwin-x64',
     'linux-x64',
     'linux-x86',
+    'linux-arm64',
+    'linux-ppc64le',
     'windows-x86'
   ].map(function (baseName) {
     const win = baseName === 'windows-x86';
@@ -42,10 +44,14 @@
     let rpmArch;
     if (name.match('linux')) {
       debArch = name.match('x86_64') ? 'amd64' : 'i386';
+      debArch = name.match('arm64') ? 'arm64' : debArch;
+      debArch = name.match('ppc64le') ? 'ppc64le' : debArch;
       debName = `kibana-${version}-${debArch}.deb`;
       debPath = resolve(rootPath, `target/${debName}`);
 
       rpmArch = name.match('x86_64') ? 'x86_64' : 'i686';
+      rpmArch = name.match('arm64') ? 'arm64' : rpmArch;
+      rpmArch = name.match('ppc64le') ? 'ppc64le' : rpmArch;
       rpmName = `kibana-${version}-${rpmArch}.rpm`;
       rpmPath = resolve(rootPath, `target/${rpmName}`);
     }
