--- kibana-5.4.1/tasks/build/os_packages.js	2017-05-27 12:53:15.000000000 +0000
+++ kibana-5.4.1/tasks/build/os_packages.js	2021-07-14 21:37:50.423345645 +0000
@@ -14,7 +14,7 @@
     grunt.file.mkdir(targetDir);
 
     config.get('platforms')
-    .filter(({ name }) => /linux-x86(_64)?$/.test(name))
+    .filter(({ name }) => /linux-(x86|x86_64|arm64|ppc64le)?$/.test(name))
     .forEach(({ buildDir, debArch, rpmArch }) => {
       const baseOptions = [
         '--force',
