diff --git a/desktop/conf.dist/hue.ini~ b/desktop/conf.dist/hue.ini
index 9b30901..a2b9bea 100644
--- a/desktop/conf.dist/hue.ini~
+++ b/desktop/conf.dist/hue.ini
@@ -95,7 +95,8 @@
 
   # Comma separated list of apps to not load at server startup.
   # e.g.: pig,zookeeper
-  ## app_blacklist=
+  # Bigtop does not bundle impala
+  app_blacklist=impala
 
   # The directory where to store the auditing logs. Auditing is disable if the value is empty.
   # e.g. /var/log/hue/audit.log
@@ -722,7 +723,8 @@
 
   # Choose whether Hue uses the GetLog() thrift call to retrieve Hive logs.
   # If false, Hue will use the FetchResults() thrift call instead.
-  ## use_get_log_api=true
+  # Bigtop Hive only supports FetchResults() 
+  use_get_log_api=false
 
   # Set a LIMIT clause when browsing a partitioned table.
   # A positive value will be set as the LIMIT. If 0 or negative, do not set any limit.
