diff --git a/build.gradle b/build.gradle
index 54b93d1792..4330fed04f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -37,6 +37,10 @@ buildscript {
     classpath "com.diffplug.spotless:spotless-plugin-gradle:$versions.spotlessPlugin"
     classpath "gradle.plugin.com.github.spotbugs.snom:spotbugs-gradle-plugin:$versions.spotbugsPlugin"
     classpath "org.gradle:test-retry-gradle-plugin:$versions.testRetryPlugin"
+    // Override jgit dependency used by grgit and spotless as we appear to auto resolve to 
+    // 5.13.0.202109080827-r and the latest version requires java 9.
+    // Requires force=true to be used in spotless as reported by ./gradlew buildEnvironment.
+    classpath("org.eclipse.jgit:org.eclipse.jgit:$versions.jgit") { force = true }
   }
 }
 
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 43c297fb7a..d73e470eb1 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -72,6 +72,7 @@ versions += [
   javassist: "3.27.0-GA",
   jetty: "9.4.43.v20210629",
   jersey: "2.34",
+  jgit: "5.12.0.202106070339-r",
   jline: "3.12.1",
   jmh: "1.27",
   hamcrest: "2.2",
