diff -ur giraph-1.1.0/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java giraph-1.1.0_/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java
--- giraph-1.1.0/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java	2014-11-01 05:25:12.000000000 +0000
+++ giraph-1.1.0_/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java	2016-12-17 17:04:28.834195182 +0000
@@ -25,7 +25,6 @@
 import org.apache.giraph.conf.ImmutableClassesGiraphConfiguration;
 import org.apache.giraph.graph.GraphMapper;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.ipc.Client;
 import org.apache.hadoop.mapreduce.Job;
 import org.apache.log4j.Logger;
 
@@ -210,7 +209,7 @@
 
     // Set the ping interval to 5 minutes instead of one minute
     // (DEFAULT_PING_INTERVAL)
-    Client.setPingInterval(giraphConfiguration, 60000 * 5);
+    giraphConfiguration.setInt("ipc.ping.interval", 60000 * 5);
 
     // Should work in MAPREDUCE-1938 to let the user jars/classes
     // get loaded first
