--- ./hbase-shell/src/main/ruby/shell/formatter.rb.orig	2017-04-22 12:39:16.617041337 +0000
+++ ./hbase-shell/src/main/ruby/shell/formatter.rb	2017-04-22 12:40:46.642890485 +0000
@@ -30,7 +30,11 @@

       def refresh_width()
         if $stdout.tty?
-          @max_width = Java::jline.Terminal.getTerminal().getTerminalWidth()
+          if Java::jline.Terminal.respond_to?(:getTerminal)
+            @max_width = Java::jline.Terminal.getTerminal.getTerminalWidth
+          else
+            @max_width = Java::jline.TerminalFactory.get.getWidth
+          end
         else
           @max_width = 0
         end
