diff --git a/Gemfile b/Gemfile
index e6b55ded0..ee2f4ef0a 100644
--- a/Gemfile
+++ b/Gemfile
@@ -25,6 +25,7 @@ gem "term-ansicolor", "~> 1.3.2", :group => :development
 gem "docker-api", "1.31.0", :group => :development
 gem "rest-client", "1.8.0", :group => :development
 gem "pleaserun", "~>0.0.28"
+gem "redis", "<= 4.4.0"
 gem "logstash-input-heartbeat"
 gem "logstash-codec-collectd"
 gem "logstash-output-xmpp"
@@ -139,3 +139,5 @@ gem "poseidon", ">= 0"
 gem "snappy", ">= 0"
 gem "webmock", "~> 1.21.0"
 gem "amazing_print", "= 1.2.1"
+gem "nokogiri", "~> 1.10.0"
+gem "msgpack", "<= 1.4.1"
