diff --git a/spark-dependencies/pom.xml b/spark-dependencies/pom.xml
index 6e7da6dc6..298e0b1ca 100644
--- a/spark-dependencies/pom.xml
+++ b/spark-dependencies/pom.xml
@@ -546,9 +546,9 @@
         <activeByDefault>true</activeByDefault>
       </activation>
       <properties>
-        <spark.version>2.2.0</spark.version>
+        <spark.version>2.2.3</spark.version>
         <protobuf.version>2.5.0</protobuf.version>
-        <py4j.version>0.10.4</py4j.version>
+        <py4j.version>0.10.7</py4j.version>
       </properties>
     </profile>

diff --git a/r/src/main/scala/org/apache/spark/api/r/RBackendHelper.scala b/r/src/main/scala/org/apache/spark/api/r/RBackendHelper.scala
index 9c1eb38d5..155f645dc 100644
--- a/r/src/main/scala/org/apache/spark/api/r/RBackendHelper.scala
+++ b/r/src/main/scala/org/apache/spark/api/r/RBackendHelper.scala
@@ -30,7 +30,7 @@ class RBackendHelper(val backend : RBackend) {
   var port : Int = 0

   def init() : Int = {
-    port = backend.init()
+    port = backend.init()._1
     port
   }

@@ -81,4 +81,4 @@ Its for testing purposes and should never be called

   def apply() : RBackendHelper = new RBackendHelper(new RBackend())

-}
\ No newline at end of file
+}
diff --git a/spark/src/main/scala/org/apache/spark/SparkRBackend.scala b/spark/src/main/scala/org/apache/spark/SparkRBackend.scala
index 05f1ac0e3..500e43f3f 100644
--- a/spark/src/main/scala/org/apache/spark/SparkRBackend.scala
+++ b/spark/src/main/scala/org/apache/spark/SparkRBackend.scala
@@ -30,7 +30,7 @@ object SparkRBackend {
   }

   def init() : Int = {
-    portNumber = backend.init()
+    portNumber = backend.init()._1
     portNumber
   }

