*** zeppelin-0.8.1/zeppelin-server/pom.xml.orig	2019-01-16 04:45:49.000000000 +0300
--- zeppelin-0.8.1/zeppelin-server/pom.xml	2019-06-18 14:00:13.047654166 +0300
***************
*** 90,95 ****
--- 90,107 ----
        <version>${project.version}</version>
        <exclusions>
          <exclusion>
+           <groupId>org.eclipse.jetty</groupId>
+           <artifactId>jetty-util</artifactId>
+         </exclusion>
+         <exclusion>
+           <groupId>org.eclipse.jetty</groupId>
+           <artifactId>jetty-webapp</artifactId>
+         </exclusion>
+         <exclusion>
+           <groupId>org.eclipse.jetty</groupId>
+           <artifactId>jetty-servlet</artifactId>
+         </exclusion>
+         <exclusion>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
          </exclusion>
***************
*** 105,110 ****
--- 117,138 ----
            <groupId>com.sun.jersey</groupId>
            <artifactId>jersey-server</artifactId>
          </exclusion>
+ 	<exclusion>
+           <groupId>com.fasterxml.jackson.core</groupId>
+           <artifactId>jackson-core</artifactId>
+         </exclusion>
+         <exclusion>
+           <groupId>com.fasterxml.jackson.core</groupId>
+           <artifactId>jackson-annotations</artifactId>
+         </exclusion>
+         <exclusion>
+           <groupId>com.nimbusds</groupId>
+           <artifactId>nimbus-jose-jwt</artifactId>
+         </exclusion>
+         <exclusion>
+           <groupId>com.fasterxml.jackson.module</groupId>
+           <artifactId>jackson-module-jaxb-annotations</artifactId>
+         </exclusion>
        </exclusions>
      </dependency>
  
***************
*** 189,195 ****
      <dependency>
        <groupId>commons-beanutils</groupId>
        <artifactId>commons-beanutils</artifactId>
!       <version>1.9.2</version>
      </dependency>
  
      <dependency>
--- 217,223 ----
      <dependency>
        <groupId>commons-beanutils</groupId>
        <artifactId>commons-beanutils</artifactId>
!       <version>1.9.3</version>
      </dependency>
  
      <dependency>
***************
*** 231,236 ****
--- 259,270 ----
        <groupId>org.eclipse.jetty.websocket</groupId>
        <artifactId>websocket-server</artifactId>
        <version>${jetty.version}</version>
+       <exclusions>
+         <exclusion>
+           <groupId>org.eclipse.jetty</groupId>
+           <artifactId>jetty-servlet</artifactId>
+         </exclusion>
+       </exclusions>
      </dependency>
  
      <dependency>
