diff --git a/security-admin/scripts/db_setup.py b/security-admin/scripts/db_setup.py
index 10acb5375..24502f4fb 100644
--- a/security-admin/scripts/db_setup.py
+++ b/security-admin/scripts/db_setup.py
@@ -149,7 +149,7 @@ def dbversionBasedOnUserName(userName):
 def set_env_val(command):
 	proc = subprocess.Popen(command, stdout = subprocess.PIPE)
 	for line in proc.stdout:
-		(key, _, value) = line.partition("=")
+		(key, _, value) = line.decode('utf8').partition('=')
 		os.environ[key] = value.rstrip()
 	proc.communicate()
 
