diff --git a/compiler/build.gradle b/compiler/build.gradle
index 60d3a43..55acee0 100644
--- a/compiler/build.gradle
+++ b/compiler/build.gradle
@@ -105,7 +105,6 @@ model {
                     // Link other (system) libraries dynamically.
                     // Clang under OSX doesn't support these options.
                     linker.args "-Wl,-Bstatic", "-lprotoc", "-lprotobuf", "-static-libgcc",
-                            "-static-libstdc++",
                             "-Wl,-Bdynamic", "-lpthread", "-s"
                 }
                 addEnvArgs("LDFLAGS", linker.args)
