diff -ur pig-0.15.0-src_/ivy/libraries.properties pig-0.15.0-src/ivy/libraries.properties
--- pig-0.15.0-src_/ivy/libraries.properties	2015-06-01 20:32:30.000000000 +0200
+++ pig-0.15.0-src/ivy/libraries.properties	2015-09-18 11:32:06.063454643 +0200
@@ -47,7 +47,7 @@
 hbase94.version=0.94.1
 hbase95.version=0.98.12-${hbase.hadoop.version}
 hsqldb.version=1.8.0.10
-hive.version=0.14.0
+hive.version=1.2.1
 httpcomponents.version=4.1
 jackson.version=1.8.8
 jackson-pig-3039-test.version=1.9.9
@@ -98,3 +98,4 @@
 leveldbjni.version=1.8
 curator.version=2.6.0
 htrace.version=3.1.0-incubating
+commons-lang3.version=3.1
diff -ur pig-0.15.0-src_/ivy.xml pig-0.15.0-src/ivy.xml
--- pig-0.15.0-src_/ivy.xml	2015-06-01 20:32:30.000000000 +0200
+++ pig-0.15.0-src/ivy.xml	2015-09-18 11:32:06.067454643 +0200
@@ -417,8 +417,6 @@
       conf="compile->master" />
     <dependency org="org.apache.hive.shims" name="hive-shims-common" rev="${hive.version}" changing="true"
       conf="compile->master" />
-    <dependency org="org.apache.hive.shims" name="hive-shims-common-secure" rev="${hive.version}" changing="true"
-      conf="compile->master" />
     <dependency org="org.apache.hive.shims" name="hive-shims-0.23" rev="${hive.version}" changing="true"
       conf="hadoop23->master" />
     <dependency org="org.apache.hive.shims" name="hive-shims-0.20S" rev="${hive.version}" changing="true"
@@ -427,6 +425,8 @@
       conf="test->master" />
     <dependency org="com.esotericsoftware.kryo" name="kryo" rev="${kryo.version}"
       conf="compile->master" />
+    <dependency org="org.apache.commons" name="commons-lang3" rev="${commons-lang3.version}"
+      conf="compile->master" />
 
     <dependency org="org.vafer" name="jdeb" rev="${jdeb.version}"
       conf="compile->master">
diff -ur pig-0.15.0-src_/src/org/apache/pig/builtin/HiveUDFBase.java pig-0.15.0-src/src/org/apache/pig/builtin/HiveUDFBase.java
--- pig-0.15.0-src_/src/org/apache/pig/builtin/HiveUDFBase.java	2015-06-01 20:32:27.000000000 +0200
+++ pig-0.15.0-src/src/org/apache/pig/builtin/HiveUDFBase.java	2015-09-18 11:32:06.067454643 +0200
@@ -28,6 +28,7 @@
 import org.apache.hadoop.hive.ql.exec.MapredContext;
 import org.apache.hadoop.hive.ql.exec.UDAF;
 import org.apache.hadoop.hive.ql.exec.UDF;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
 import org.apache.hadoop.hive.ql.udf.generic.Collector;
 import org.apache.hadoop.hive.ql.udf.generic.GenericUDAFResolver;
 import org.apache.hadoop.hive.ql.udf.generic.GenericUDF;
@@ -103,7 +104,12 @@
         String className = funcName;
         Class udfClass;
         if (FunctionRegistry.getFunctionNames().contains(funcName)) {
-            FunctionInfo func = FunctionRegistry.getFunctionInfo(funcName);
+            FunctionInfo func;
+            try {
+                func = FunctionRegistry.getFunctionInfo(funcName);
+            } catch (SemanticException e) {
+                throw new IOException(e);
+            }
             udfClass = func.getFunctionClass();
         } else {
             udfClass = PigContext.resolveClassName(className);
diff -ur pig-0.15.0-src_/test/org/apache/pig/test/TestLoaderStorerShipCacheFiles.java pig-0.15.0-src/test/org/apache/pig/test/TestLoaderStorerShipCacheFiles.java
--- pig-0.15.0-src_/test/org/apache/pig/test/TestLoaderStorerShipCacheFiles.java	2015-06-01 20:32:30.000000000 +0200
+++ pig-0.15.0-src/test/org/apache/pig/test/TestLoaderStorerShipCacheFiles.java	2015-09-18 11:32:06.067454643 +0200
@@ -50,10 +50,9 @@
             hadoopVersion = "23";
         }
         String[] expectedJars = new String[] {"hive-common", "hive-exec", "hive-serde", 
-                "hive-shims-0." + hadoopVersion, "hive-shims-common-0", "hive-shims-common-secure",
-                "kryo"};
+                "hive-shims-0." + hadoopVersion, "hive-shims-common", "kryo"};
 
-        checkPlan(pp, expectedJars, 7, pigServer.getPigContext());
+        checkPlan(pp, expectedJars, 6, pigServer.getPigContext());
     }
 
     @Test
@@ -67,10 +66,9 @@
             hadoopVersion = "23";
         }
         String[] expectedJars = new String[] {"hive-common", "hive-exec", "hive-serde", 
-                "hive-shims-0." + hadoopVersion, "hive-shims-common-0", "hive-shims-common-secure",
-                "kryo"};
+                "hive-shims-0." + hadoopVersion, "hive-shims-common", "kryo"};
 
-        checkPlan(pp, expectedJars, 7, pigServer.getPigContext());
+        checkPlan(pp, expectedJars, 6, pigServer.getPigContext());
     }
 
     @Test
Nur in pig-0.15.0-src/test/org/apache/pig/test: TestLoaderStorerShipCacheFiles.java.orig.
